api/src/test/java/com/epam/pipeline/acl/contextual/ContextualPreferenceApiServiceTest.java

```diff
@@ -0,0 +1,138 @@
+import com.epam.pipeline.test.creator.contextual.ContextualPreferenceCreatorUtils;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING_LIST;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+public class ContextualPreferenceApiServiceTest extends AbstractAclTest {
+
+    @Autowired
+    private ContextualPreferenceApiService preferenceApiService;
+
+    @Autowired
+    private ContextualPreferenceManager mockPreferenceManager;
+
+    private final ContextualPreference contextualPreference =
+            ContextualPreferenceCreatorUtils.getContextualPreference();
+
+    private final List<ContextualPreference> preferenceList = Collections.singletonList(contextualPreference);
+
+    private final ContextualPreferenceExternalResource cpeResource =
+            ContextualPreferenceCreatorUtils.getCPExternalResource();
+
+    private final ContextualPreferenceVO contextualPreferenceVO =
+            ContextualPreferenceCreatorUtils.getContextualPreferenceVO();
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadAllContextualPreferencesForAdmin() {
+        doReturn(preferenceList).when(mockPreferenceManager).loadAll();
+
+        List<ContextualPreference> resultPreferenceList = preferenceApiService.loadAll();
+
+        assertThat(resultPreferenceList.size()).isEqualTo(1);
+        assertThat(resultPreferenceList.get(0)).isEqualTo(contextualPreference);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldDenyLoadingAllContextualPreferencesForNotAdmin() {
+        doReturn(preferenceList).when(mockPreferenceManager).loadAll();
+
+        preferenceApiService.loadAll();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadContextualPreferenceForAdmin() {
+        doReturn(contextualPreference).when(mockPreferenceManager).load(TEST_STRING, cpeResource);
+
+        ContextualPreference resultPreference = preferenceApiService.load(TEST_STRING, cpeResource);
+
+        assertThat(resultPreference).isEqualTo(contextualPreference);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldDenyLoadingContextualPreferenceForNotAdmin() {
+        doReturn(contextualPreference).when(mockPreferenceManager).load(TEST_STRING, cpeResource);
+
+        preferenceApiService.load(TEST_STRING, cpeResource);
+    }
+
+    @Test
+    public void shouldSearchContextualPreference() {
```

Comment:
We expect here to be at least a simple user role here so let's add corresponding `@WithMockUser` annotation.